File: Sources/FitUI/Components/FitCard.swift
import SwiftUI

// MARK: - FitCard
//
// Content container for grouped information. Optional header with
// edit action. Edge: light = soft drop shadow (no border); dark = surface
// bg on a darker screen, tonal contrast only (no border). Keeping dark
// borderless avoids a hard hairline flashing on theme toggle.
// See `docs/components.md`.

public struct FitCard<Content: View>: View {
    let title: String?
    let onEdit: (() -> Void)?
    let content: () -> Content

    @Environment(\.fitTheme) private var theme

    public init(
        title: String? = nil,
        onEdit: (() -> Void)? = nil,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.title = title
        self.onEdit = onEdit
        self.content = content
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: FitSpacing.sp3) {
            if let title = title {
                HStack {
                    Text(title)
                        .font(.custom(FitFont.family, size: 18).weight(.medium))
                        .foregroundColor(theme.textPrimary)
                    Spacer()
                    if let edit = onEdit {
                        Button(action: edit) {
                            Image(systemName: "pencil")
                                .font(.system(size: FitSize.iconMd))
                                .foregroundColor(theme.textTertiary)
                        }
                        .buttonStyle(.plain)
                    }
                }
            }
            content()
        }
        .padding(FitSpacing.sp5)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(theme.cardBg)
        .clipShape(RoundedRectangle(cornerRadius: FitRadius.card))
        .shadow(color: theme.screenBg == Color(hex: "F2F2F7") ? Color.black.opacity(0.07) : .clear,
                radius: 12, x: 0, y: 0)
    }
}

// MARK: - FitCardRow — single info row with icon + text

public struct FitCardRow: View {
    let icon: String
    let text: String

    @Environment(\.fitTheme) private var theme

    public init(icon: String, text: String) {
        self.icon = icon
        self.text = text
    }

    public var body: some View {
        HStack(spacing: FitSpacing.sp2) {
            Image(systemName: icon)
                .font(.system(size: FitSize.iconMd))
                .foregroundColor(theme.textSecondary)
                .frame(width: 16, height: 16)

            Text(text)
                .font(FitFont.body2)
                .foregroundColor(theme.textSecondary)
        }
    }
}

File: android/src/main/kotlin/com/fit321/fitui/components/FitCard.kt
package com.fit321.fitui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import com.fit321.fitui.tokens.FitElevation
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Shape
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitColors
import com.fit321.fitui.tokens.FitFont
import com.fit321.fitui.tokens.FitRadius
import com.fit321.fitui.tokens.FitSize
import com.fit321.fitui.tokens.FitSpacing

/**
 * Content container for grouped information. Optional header with edit
 * action. Mirrors Swift `FitCard`. See `docs/components.md`.
 *
 * Surface = `theme.surfaceDefault`. Edge: light = soft drop shadow only (no
 * border); dark = tonal contrast only (no border). Keeping both borderless
 * avoids a hard hairline flashing on theme toggle. Components must never
 * hardcode colors — wrap with `FitTheme { FitCard { ... } }`.
 */
@Composable
fun FitCard(
    modifier: Modifier = Modifier,
    shape: Shape = RoundedCornerShape(FitRadius.card),
    surface: Color? = null,
    contentPadding: PaddingValues = PaddingValues(FitSpacing.sp5),
    verticalArrangement: Arrangement.Vertical = Arrangement.spacedBy(FitSpacing.sp3),
    title: String? = null,
    onEdit: (() -> Unit)? = null,
    onClick: (() -> Unit)? = null,
    content: @Composable () -> Unit
) {
    val theme = LocalFitTheme.current
    val isLight = theme.screenBg != FitColors.Gray.g900

    val clickableMod = if (onClick != null) Modifier.clickable { onClick() } else Modifier

    Column(
        modifier = modifier
            .fillMaxWidth()
            .then(with(FitElevation) { Modifier.fitCardElevation(isLight, shape) })
            .clip(shape)
            .background(surface ?: theme.surfaceDefault)
            .then(clickableMod)
            .padding(contentPadding),
        verticalArrangement = verticalArrangement
    ) {
        if (title != null) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    text = title,
                    color = theme.textPrimary,
                    style = FitFont.heading3
                )
                Spacer(modifier = Modifier.weight(1f))
                if (onEdit != null) {
                    Icon(
                        imageVector = Icons.Default.Edit,
                        contentDescription = null,
                        tint = theme.textTertiary,
                        modifier = Modifier
                            .size(FitSize.iconMd)
                            .clickable { onEdit() }
                    )
                }
            }
        }
        content()
    }
}

/**
 * Single info row inside a `FitCard` — leading icon (16dp) + secondary text.
 * Mirrors Swift `FitCardRow`.
 */
@Composable
fun FitCardRow(
    icon: ImageVector,
    text: String,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current
    Row(
        modifier = modifier,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = theme.textSecondary,
            modifier = Modifier.size(FitSize.iconMd)
        )
        Spacer(modifier = Modifier.width(FitSpacing.sp2))
        Text(
            text = text,
            color = theme.textSecondary,
            style = FitFont.body2
        )
    }
}
